src/prototype/WingetCreateMakeDSC.ps1

```diff
@@ -0,0 +1,63 @@
+# This script is a prototype for quickly creating DSC files.
+
+if ($null -eq (Get-InstalledModule -Name Microsoft.Winget.Client))
+{
+  Install-Module Microsoft.Winget.Client
+}
+
+if ($null -eq (Get-InstalledModule -Name powershell-yaml))
+{
+  Install-Module powershell-yaml
+}
+
+[System.Collections.ArrayList]$finalPackages = @()
+$configurationVersion = "0.2.0" 
+
+$continue = $true
+while ($continue)
+{
+  $appId = Read-Host "What is the id of your app?"
+  $findResult = Find-WinGetPackage $appId
+  
+  if ($findResult.count -ne 0)
+  {
+    $index=0
+    foreach ($package in $findResult)
+    {
+        $packageDetails = "[$($index)] $($package.Name) | $($package.Id) | $($package.Version)"
+        Write-Host $packageDetails
+        $index++
+    }
+
+    $selection = -1
+    $packageSelected = $false
+    while (-not($packageSelected))
+    {
+      $selection = [int](Read-Host "Input the number of the package you want to select")
+      if (($selection -gt $findResult.count) -or ($selection -lt 0))
```

Comment:
`$selection -gt $findResult.count` will incorrectly allow one greater than the last value to be a valid selection with the current logic. I would suggest keeping this check as is but do the following:
- initiating $index from 1 (since it's a user-facing number) 
- change ($selection -lt 0) to `($selection -lt 1)`
- change `$selectedPackage = $findResult[$selection]` to be `$selectedPackage = $findResult[$selection - 1]`

Reply:
With my above suggestion adding the index as a property to each package, this would also be simplified
* `if $selection -notin $findResult.Index` can be used to check if the index is in range
* `$selectedPackage = $findResult.Where({$_.Index -eq $selection})` can be used to set the selected package

Reply:
I went with @mdanish-kh  proposal. I played around with Trenly's but I fell over.  :)

Reply:
I'll take a look again and might make a PR to your fork if it would be beneficial